compose/utils.py

```diff
@@ -91,3 +93,15 @@ def json_hash(obj):
     h = hashlib.sha256()
     h.update(dump)
     return h.hexdigest()
+
+
+def interpolate(string, mapping):
+    try:
+        return Template(string).substitute(defaultdict(lambda: "", mapping))
+    except ValueError:
```

Comment:
Oh, I just realized that is what defaultdict is handling.  So the result of a missing value would be empty string instead of leaving the original string. That should work.